### PR TITLE
init shine, and ocamlPackages.shine

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Description of changes
Part of splitting up #140777 for easier review

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/shine/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    rev = "2e1de686ea031f1056df389161ea2b721bfdb39e";
+    sha256 = "0v6i4ym5zijki6ffkp2qkp00lk4fysjhmg690xscj23gwz4zx8ir";
+  };
+
+  buildInputs = [ dune-configurator ];
+  propagatedBuildInputs = [ shine ];
+
+  meta = with lib; {
+    homepage = "https://github.com/savonet/ocaml-shine";
+    description = "binding to the fixed-point mp3 encoding library shine. ";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/shine/default.nix:21]: Please double check the relevant part of the manual: https://nixos.org/manual/nixpkgs/stable/#var-meta-description

File: pkgs/development/libraries/shine/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchFromGitHub {
+    owner = "toots";
+    repo = "shine";
+    rev = version;
+    sha256 = "06nwylqqji0i1isdprm2m5qsdj4qiywcgnp69c5b55pnw43f07qg";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
```
Comment (vbgl) [inline pkgs/development/libraries/shine/default.nix:15]: If you get the sources from the released tarball at <https://github.com/toots/shine/releases/download/3.1.1/shine-3.1.1.tar.gz>, you won’t need the `autoreconfHook`.

Comment (dali99) [reply pkgs/development/libraries/shine/default.nix:15]: Is that really the preference in NixOS? I figured we'd always want to build from the "actual source" and not the processed sources

Comment (vbgl) [reply pkgs/development/libraries/shine/default.nix:15]: I don’t know.

**Pull request merged**